packages/components/src/components/Log/Log.js

```diff
@@ -28,17 +30,147 @@ const itemSize = 15; // This should be kept in sync with the line-height in SCSS
 const defaultHeight = itemSize * 100 + itemSize / 2;
 
 export class LogContainer extends Component {
-  state = { loading: true };
+  constructor(props) {
+    super(props);
+    this.state = { loading: true };
+    this.logRef = React.createRef();
+    this.textRef = React.createRef();
+  }
 
   componentDidMount() {
     this.loadLog();
+    window.addEventListener("scroll", this.handleLogScroll, true);
+    if (this.shouldAutoScroll(this.props.stepStatus)) {
+      this.scrollToBottomLog();
+      return; // the scroll event listener will call handleLogScroll
+    }
+    this.handleLogScroll();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      prevState.logs?.length !== this.state.logs?.length ||
+      prevProps.isLogsMaximized !== this.props.isLogsMaximized
+    ) {
+      if (this.shouldAutoScroll(this.props.stepStatus, prevProps.stepStatus)) {
+        this.scrollToBottomLog();
+        return; // the scroll event listener will call handleLogScroll
+      }
+      this.handleLogScroll();
+    }
   }
 
   componentWillUnmount() {
     clearInterval(this.timer);
+    window.removeEventListener("scroll", this.handleLogScroll, true);
     this.cancelled = true;
   }
 
+  handleLogScroll = () => {
+    this.updateScrollButtonHorizontalCoordinate();
+    const isLogTopUnseen = this.isLogTopUnseen();
+    const isLogBottomUnseen = this.isLogBottomUnseen();
+
+    if (
+      isLogBottomUnseen !== this.state.isLogBottomUnseen ||
+      isLogTopUnseen !== this.state.isLogTopUnseen
+    ) {
+      this.setState({
+        isLogBottomUnseen,
+        isLogTopUnseen
+      });
+    }
+  };
+  
+  isLogBottomUnseen = () => {
+    return utils.isElementEndBelowViewBottom(this.logRef?.current)
+      || utils.hasElementPositiveScrollBottom(this.textRef?.current?.firstElementChild);
+  };
+
+  isLogTopUnseen = () => {
+    return utils.isElementStartAboveViewTop(this.logRef?.current)
+      || utils.hasElementPositiveScrollTop(this.textRef?.current?.firstElementChild);
+  };
+
+  scrollToBottomLog = () => {
+    const longTextElement = this.textRef?.current?.firstElementChild;
+    if (utils.hasElementPositiveScrollBottom(longTextElement)) {
+      longTextElement.scrollTop = longTextElement.scrollHeight - longTextElement.clientHeight;
+    }
+    const rootElement = document.documentElement;
+    rootElement.scrollTop = rootElement.scrollHeight - rootElement.clientHeight;
```

Comment:
rather than setting the `scrollTop` directly can we use the `scrollTo` API instead?
i.e. `rootElement.scrollTo({ top: … })`, this also lets us control the scroll behaviour (smooth / auto).